src/DocRaptor.php

```diff
@@ -39,15 +40,17 @@ class ApiWrapper
     protected $baseurl;
 
 
-
     /**
+     * @param HttpTransferInterface $httpClient
      * @param string|null $api_key
      */
-    public function __construct($api_key = null)
+    public function __construct(HttpTransferInterface $httpClient, $api_key = null)
```

Comment:
Ah. I see.

I guess the choice comes down to what you prefer:

```php
(new ApiWrapper("api key")).fetchDocument($filename)
```
*or*
```php
(new ApiWrapper(null, "api key")).fetchDocument($filename)
```

I think the first is much nicer.

---
I guess the other way is something like:
```php
$drClient = new SynchronousHttpClient() # or Asynchronous
(new ApiWrapper($drClient, "api key")).fetchDocument($filename)
```

which is nice if we wanted to differentiate how to handle the requests just using the client.
---
Thoughts?